### PR TITLE
asset: make it easier to load and write asset out of rendering phase

### PR DESCRIPTION
**Pull request by xiang90**
/cc @diegs 

This makes it easier to load in recovery etcd template and output recovery etcd manifest as a static pod.

### CODE REVIEW

File: pkg/asset/asset.go [hunk 1]
```diff
@@ -185,14 +185,21 @@ func (as Assets) WriteFiles(path string) error {
...
 	return nil
 }
+
+func (a Asset) WriteFile(path string) error {
+	f := filepath.Join(path, a.Name)
+	if err := os.MkdirAll(filepath.Dir(f), 0755); err != nil {
+		return err
+	}
+	fmt.Printf("Writing asset: %s\n", f)
+	if err := ioutil.WriteFile(f, a.Data, 0600); err != nil {
```
Comment (diegs) [inline pkg/asset/asset.go]: nit: return `ioutil.WriteFile(...)`?

File: pkg/asset/k8s.go [hunk 2]
```diff
@@ -163,7 +163,7 @@ func secretFromAssets(name, namespace string, assetNames []string, assets Assets
 	})
 }
 
-func mustCreateAssetFromTemplate(name string, template []byte, data interface{}) Asset {
+func MustCreateAssetFromTemplate(name string, template []byte, data interface{}) Asset {
```
Comment (diegs) [inline pkg/asset/k8s.go]: Can you elaborate on why you need to make this a public API?

Currently in bootkube recovery tool we simply return an Assets array, and then the caller invokes `Assets.WriteFiles()` to write them out to disk.

Comment (xiang90) [inline pkg/asset/k8s.go]: Our templates sit inside an internal pkg. There is no easy way to ONLY load in one recovery etcd template. Alternatively, I can put a special loading logic inside Asset pkg (WriteOutRecoveryEtcd or something). But I feel it is less idea.

Also I actually want to put the recovery etcd template inside the recovery pkg.

Comment (xiang90) [inline pkg/asset/k8s.go]: > bootkube recovery tool 

One difference is that we load the actual manifest bytes from etcd directly. We do not try to generate manifest bytes from a template. So the current recovery tool does not need to public `MustCreateAssetFromTemplate` func.

Comment (diegs) [inline pkg/asset/k8s.go]: Ok, SGTM

**Comment by xiang90** [comment]
@diegs Thanks for the quick review. The nit is fixed. I will get it merged on green.

**Pull request merged**